File: AtlasWeb/Core/Modules/Saglik_Lojistik/Stok_Modulleri/Sihhiye_Modulleri/Tasinir_Mal_Islem_Modulu/ChattelDocumentOutputWithAccountancyNewFormViewModel.cs
//$DCFA882E
using System;
using System.Linq;
using TTInstanceManagement;
using Core.Models;
using TTObjectClasses;
using TTDefinitionManagement;
using TTUtils;
using TTDataDictionary;

namespace Core.Controllers
{
    public partial class ChattelDocumentOutputWithAccountancyServiceController
    {
        partial void AfterContextSaveScript_ChattelDocumentOutputWithAccountancyNewForm(ChattelDocumentOutputWithAccountancyNewFormViewModel viewModel, ChattelDocumentOutputWithAccountancy chattelDocumentOutputWithAccountancy, TTDefinitionManagement.TTObjectStateTransitionDef transDef, TTObjectContext objectContext)
        {
            string sonucMesaji = "";
            try
            {
                if (viewModel._ChattelDocumentOutputWithAccountancy.CurrentStateDefID == ChattelDocumentOutputWithAccountancy.States.Completed)
                {


                    if (TTObjectClasses.SystemParameter.GetParameterValue("SENDTOMKYSFORAFTERCONTEXT", "TRUE") == "TRUE")
                    {
                        sonucMesaji = chattelDocumentOutputWithAccountancy.SendMKYSForOutputDocument(Common.CurrentResource.MkysPassword);
                        TTUtils.InfoMessageService.Instance.ShowMessage(sonucMesaji);
                    }
                }
            }
            catch (Exception ex)
            {
                throw new Exception(ex.Message);
            }


            sonucMesaji = "";
            foreach (var detail in chattelDocumentOutputWithAccountancy.ChattelDocumentOutputDetailsWithAccountancy.Select(string.Empty))
            {
                foreach (var transaction in detail.StockTransactions)
                {
                    // Alma bildirimi yapýldýysa iptaili için eklendi
                    if (string.IsNullOrEmpty(transaction.ReceiveNotificationID) == false)
                    {
                        try
                        {
                            transaction.DeliveryNotifictionID = StockAction.UTSMakeDeliveryNotification(transaction);
                            sonucMesaji += transaction.DeliveryNotifictionID;
                            objectContext.Save();
                        }
                        catch (Exception ex)
                        {
                            throw new Exception(ex.Message);
                        }
                    }
                }
            }
            if (String.IsNullOrEmpty(sonucMesaji) == false)
                TTUtils.InfoMessageService.Instance.ShowMessage(sonucMesaji + " Üts verme bildirimleri yapýldý");

        }
    }
}

namespace Core.Models
{
    public partial class ChattelDocumentOutputWithAccountancyNewFormViewModel
    {
        public OuttableLotDTO[] OuttableLotList { get; set; }
        public QRCodeReadDTO[] QRCodeReadList { get; set; }
    }

    public class OuttableLotDTO
    {
        public Currency Amount { get; set; }
        public Currency RestAmount { get; set; }
        public string BudgetTypeName { get; set; }
        public DateTime ExpirationDate { get; set; }
        public string LotNo { get; set; }
        public string SerialNo { get; set; }
        public Guid TrxObjectID { get; set; }
        public Int16 StockActionDetailOrderNo { get; set; }
    }

    public class QRCodeReadDTO
    {
        public string Barcode { get; set; }
        public string LotNo { get; set; }
        public DateTime ExpirationDate { get; set; }
        public string SerialNo { get; set; }
        public Currency Amount { get; set; }
        public Guid MaterialObjectID { get; set; }
    }
}

File: AtlasWeb/Core/Modules/Saglik_Lojistik/Stok_Modulleri/Sihhiye_Modulleri/Tasinir_Mal_Islem_Modulu/ChattelDocumentOutputWithAccountancyNewFormViewModel.ts
//$4FC6001C
import { BaseViewModel } from "NebulaClient/Model/BaseViewModel";
import { ChattelDocumentOutputWithAccountancy } from 'NebulaClient/Model/AtlasClientModel';
import { ChattelDocumentOutputDetailWithAccountancy } from 'NebulaClient/Model/AtlasClientModel';
import { StockActionSignDetail } from 'NebulaClient/Model/AtlasClientModel';
import { Store } from 'NebulaClient/Model/AtlasClientModel';
import { Material } from 'NebulaClient/Model/AtlasClientModel';
import { StockCard } from 'NebulaClient/Model/AtlasClientModel';
import { DistributionTypeDefinition } from 'NebulaClient/Model/AtlasClientModel';
import { StockLevelType } from 'NebulaClient/Model/AtlasClientModel';
import { Accountancy } from 'NebulaClient/Model/AtlasClientModel';
import { ResUser } from 'NebulaClient/Model/AtlasClientModel';

import { Type } from 'NebulaClient/ClassTransformer';
import { Guid } from "app/NebulaClient/Mscorlib/Guid";

export class ChattelDocumentOutputWithAccountancyNewFormViewModel extends BaseViewModel {
    @Type(() => ChattelDocumentOutputWithAccountancy)
    public _ChattelDocumentOutputWithAccountancy: ChattelDocumentOutputWithAccountancy = new ChattelDocumentOutputWithAccountancy();
    @Type(() => ChattelDocumentOutputDetailWithAccountancy)
    public ChattelDocumentDetailsWithAccountancyGridList: Array<ChattelDocumentOutputDetailWithAccountancy> = new Array<ChattelDocumentOutputDetailWithAccountancy>();
    @Type(() => StockActionSignDetail)
    public StockActionSignDetailsGridList: Array<StockActionSignDetail> = new Array<StockActionSignDetail>();
    @Type(() => Store)
    public Stores: Array<Store> = new Array<Store>();
    @Type(() => Material)
    public Materials: Array<Material> = new Array<Material>();
    @Type(() => StockCard)
    public StockCards: Array<StockCard> = new Array<StockCard>();
    @Type(() => DistributionTypeDefinition)
    public DistributionTypeDefinitions: Array<DistributionTypeDefinition> = new Array<DistributionTypeDefinition>();
    @Type(() => StockLevelType)
    public StockLevelTypes: Array<StockLevelType> = new Array<StockLevelType>();
    @Type(() => Accountancy)
    public Accountancys: Array<Accountancy> = new Array<Accountancy>();
    @Type(() => ResUser)
    public ResUsers: Array<ResUser> = new Array<ResUser>();
    public OuttableLotList: Array<OuttableLotDTO> = new Array<OuttableLotDTO>();
    public QRCodeReadList: Array<QRCodeReadDTO> = new Array<QRCodeReadDTO>();
}

export class OuttableLotDTO {
    public Amount: number;
    public RestAmount: number;
    public BudgetTypeName: string;
    public ExpirationDate: Date;
    public LotNo: string;
    public SerialNo: string;
    @Type(() => Guid)
    public TrxObjectID: Guid;
    public StockActionDetailOrderNo: number;
}

export class QRCodeReadDTO {
    public Barcode: string;
    public LotNo:string;
    public ExpirationDate: Date;
    public SerialNo: string;
    public Amount: number;
    @Type(() => Guid)
    public MaterialObjectID: Guid;
}
